### PR TITLE
feature/v1-076/employee profile for manager

### PR DESCRIPTION
**Pull request by Julia-Maliutsina**
1. Page with info about employee,their skills and courses at '/employees/:employeeId'.
2. API for getting employee info.
3. Responsive design for employee page.

### CODE REVIEW

File: frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+export const SearchCourse = styled(Input)({
+  [theme.breakpoints.up('xs')]: {
+    fontSize: '16px!important',
+    lineHeight: '33.61px',
+  },
+  [theme.breakpoints.up('sm')]: {
+    fontSize: '18px!important',
+  },
+});
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts:23]:
```suggestion
export const SearchCourse = styled(Input)({
   fontSize: '18px!important',
  [theme.breakpoints.down('sm')]: {
    fontSize: '16px!important',
    lineHeight: '33.61px',
  },
});
```

File: frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+export const CoursesBox = styled('div')({
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts:25]:
```suggestion
export const CoursesBox = styled('div')({
    maxWidth: '100%',
    marginLeft: '0',
    marginRight: '0',
  [theme.breakpoints.down('xl')]: {
    maxWidth: '828px',
    marginLeft: '54px',
    marginRight: '47px',
  },
  [theme.breakpoints.down('md')]: {
    marginLeft: '54px',
    marginRight: '19px',
  },
  [theme.breakpoints.down('sm')]: {
    maxWidth: '288px',
    marginLeft: '13px',
    marginRight: '19px',
  },
});
```

File: frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+export const CoursesListItem = styled('div')({
+  display: 'flex',
+  alignItems: 'center',
+  width: '350px',
+  height: '80px',
+  background: '#F5F5F5',
+  border: '1px solid #EFEFEF',
+  borderRadius: '5px',
+  paddingTop: '16px',
+  fontFamily: '"Ubuntu", sans-serif',
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts:70]: Is it necessary? We have already declared it in theme

File: frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+});
+
+export const CoursesDivider = styled(Divider)({
+  [theme.breakpoints.down('sm')]: {
+    display: 'none',
+  },
+  [theme.breakpoints.up('sm')]: {
+    display: 'block',
+  },
+});
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/EmployeeCourses/styled.ts:119]:
```suggestion
export const CoursesDivider = styled(Divider)({
   display: 'block',
  [theme.breakpoints.down('sm')]: {
    display: 'none',
  },
});
```

File: frontend/src/pages/EmployeeProfile/EmployeeProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  SkillsAndCoursesButton,
+  AddCourseButton,
+  UserSkillsWrapper,
+  UserInfoListItems,
+  ExpandMore,
+  ExpandLess,
+} from './styled';
+
+interface Props {
+  employeeCourses?: ClientCourse[];
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/EmployeeProfile.tsx:34]: fix order please

File: frontend/src/pages/EmployeeProfile/styled.ts [hunk 6]
```diff
@@ -0,0 +1,253 @@
+import styled from 'styled-components';
+import { List, ListItem, Button as ToggleButton, ButtonGroup } from '@mui/material';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ExpandLessIcon from '@mui/icons-material/ExpandLess';
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/styled.ts:4]:
```suggestion
import { ExpandMoreIcon, ExpandLessIcon } from '@mui/icons-material';
```

File: frontend/src/pages/EmployeeProfile/styled.ts [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+interface ShowOnMobile {
+  displayInfo: boolean;
+}
+
+export const ProfileBox = styled('div')({
+  width: '90%',
+  display: 'flex',
+  flexDirection: 'row',
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/styled.ts:20]: 'row' is a default value of flex-direction, so you can remove it

File: frontend/src/pages/EmployeeProfile/styled.ts [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+  margin: '24px 0px 92px 165px',
+  [theme.breakpoints.down('xl')]: {
+    margin: '8px 0px 50px 112px !important',
+  },
+  [theme.breakpoints.down('md')]: {
+    margin: '16px 0px 50px 16px !important',
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/styled.ts:26]:
```suggestion
    margin: '16px 0 50px 16px !important',
```

File: frontend/src/pages/EmployeeProfile/styled.ts [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+    width: '100%',
+  },
+});
+
+export const AvatarWrapper = styled('div')({
+  textAlign: 'center',
+  marginRight: '10%',
+  marginLeft: '0px',
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/styled.ts:34]: for zero lengths the unit identifier is optional, **0px === 0em === 0%**

File: frontend/src/pages/EmployeeProfile/styled.ts [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+    width: '100px',
+  },
+  [theme.breakpoints.down('lg')]: {
+    width: '60px',
+    marginRight: '20px',
+  },
+});
+
+export const EmployeeName = styled(ListItem)({
+  fontFamily: '"Ubuntu", sans-serif',
```
Comment (irina-sinitsyna) [inline frontend/src/pages/EmployeeProfile/styled.ts:46]: is it necessary here?

**Review by irina-sinitsyna — CHANGES_REQUESTED** [review]

**Review by irina-sinitsyna — DISMISSED** [review]

File: frontend/src/pages/EmployeeProfile/EmployeeCourses/EmployeeCourses.tsx [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+              <CoursesListItem>
+                <ProgressBar size={SIZE.small} value={countProgress(course.progress)} />
+                <CourseItemText>
+                  <CourseTitle>{course.course.title}</CourseTitle>
+                  <CourseStatus>
+                    {course.status[0].toUpperCase() + course.status.slice(1)}
+                  </CourseStatus>
+                </CourseItemText>
+              </CoursesListItem>
+              {id !== coursesArray.length - PREVIOUS_ITEM && <CoursesDivider />}
```
Comment (k-krasilnikova) [inline frontend/src/pages/EmployeeProfile/EmployeeCourses/EmployeeCourses.tsx:84]: What is this calculation mean? Move it to constant please

File: frontend/src/pages/EmployeeProfile/EmployeeCourses/EmployeeCoursesContainer.tsx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+            .toLocaleLowerCase()
+            .includes(searchCourse.trimEnd().toLocaleLowerCase()) ||
+          course.course.title.toLowerCase().includes(searchCourse.toLowerCase()),
+      );
+      setEmployeeCourses(filteredCourses);
+    }
+  }, [searchCourse, courses]);
+
+  const searchCourseInList = (value: string) => {
+    const formattedValue = value.split(/\s+/).join(' ').trimStart();
```
Comment (k-krasilnikova) [inline frontend/src/pages/EmployeeProfile/EmployeeCourses/EmployeeCoursesContainer.tsx:29]: !

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]

File: frontend/src/pages/EmployeeProfile/EmployeeProfile.tsx [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+      </UserInfoList>
+    </ProfileBox>
+    <SkillsAndCoursesBox>
+      <EmployeeButtonGroup variant="contained">
+        <SkillsAndCoursesButton
+          isOpened={
+            (employeeInfo === EMPLOYEE_INFO.skills && !hoveredButton) ||
+            hoveredButton === EMPLOYEE_INFO.skills
+          }
+          onClick={() => toggleEmployeeInfo(EMPLOYEE_INFO.skills)}
```
[the remark this answers is not in the document]
Comment (irina-sinitsyna) [reply frontend/src/pages/EmployeeProfile/EmployeeProfile.tsx:108]: @k-krasilnikova Got it :+1:

File: frontend/src/pages/EmployeeProfile/EmployeeCourses/EmployeeCourses.tsx (hunk 11 above)
Comment (Julia-Maliutsina) [reply frontend/src/pages/EmployeeProfile/EmployeeCourses/EmployeeCourses.tsx:84]: It's not a calculation, it just adds Divider at the end of each element except for the last one. 
Replaced it with  {id < coursesArray.length - 1 && <CoursesDivider />} to be more clear

File: frontend/src/App.tsx [hunk 14]
```diff
@@ -49,7 +50,10 @@ const App: React.FC = () => (
               <Route path=":courseId" element={<DetailedCourse page={PAGES.coursesList} />} />
             </Route>
             <Route path={PATHS.help} element={<Help />} />
-            <Route path={PATHS.employees} element={<Employees />} />
+            <Route path={PATHS.employees}>
+              <Route index element={<Employees />} />
+              <Route path=":employeeId" element={<EmployeeProfile />} />
```
Comment (k-krasilnikova) [inline frontend/src/App.tsx:55]: Move path please to the constant

**Review by k-krasilnikova — COMMENTED** [review]
Almost done!
One more fix required

File: frontend/src/App.tsx (hunk 14 above)
Comment (Julia-Maliutsina) [reply frontend/src/App.tsx]: Done

**Pull request closed without merging**